core-plugins/src/test/java/io/cdap/plugin/batch/joiner/JoinerTestRun.java

```diff
@@ -216,6 +228,30 @@ public void testOuterJoin() throws Exception {
                                    "requiredInputs", "film,filmActor"),
                                  null));
 
+    ETLStage joinSinkStage = new ETLStage(
+      "sink", new ETLPlugin("TPFSAvro", BatchSink.PLUGIN_TYPE,
+                            ImmutableMap.of(Properties.TimePartitionedFileSetDataset.SCHEMA, outputSchema.toString(),
+                                            Properties.TimePartitionedFileSetDataset.TPFS_NAME, joinedDatasetName),
+                            null));
+
+    joinHelper(filmStage, filmActorStage, filmCategoryStage, joinStage, joinSinkStage,
+               filmDatasetName, filmActorDatasetName, filmCategoryDatasetName, joinedDatasetName,
+               outputSchema, this::verifyOuterJoinOutput);
+  }
+
+  @Test
+  public void testOuterJoinWithUnknownInputSchemas() throws Exception {
+    String filmDatasetName = "film-outerjoin-unknown-inputschemas";
+    String filmCategoryDatasetName = "film-category-outerjoin-unknown-inputschemas";
+    String filmActorDatasetName = "film-actor-outerjoin-unknown-inputschemas";
+    String joinedDatasetName = "outerjoin-output-unknown-inputschemas";
+
+
+    ETLStage filmStage = new ETLStage("film", MockSource.getPlugin(filmDatasetName, FILM_SCHEMA));
```

Comment:
the schemas are being set here, isn't this test supposed to be for null input schemas? You should be using the method that doesn't take a schema as an argument.

Reply:
thanks for catching that! fixed.